src/Reflection/FunctionReflection.php

```diff
@@ -36,4 +36,6 @@ public function getAsserts(): Assertions;
 
 	public function getDocComment(): ?string;
 
+	public function returnsByReference(): ?bool;
```

Comment:
I went with `?bool` so we can differentiate between whether we know that a function/method returns by reference, or `null` if it is unknown